packages/legacy/core/App/container-impl.ts

```diff
@@ -43,12 +54,48 @@ export class MainContainer implements Container {
       }
     )
 
+    this.container.registerInstance(TOKENS.LOAD_STATE, async (dispatch: React.Dispatch<ReducerAction<unknown>>) => {
+      const loadState = async <Type>(key: LocalStorageKeys, updateVal: (newVal: Type) => void) => {
+        const data = await AsyncStorage.getItem(key)
+        if (data) {
+          const dataAsJSON = JSON.parse(data) as Type
+          updateVal(dataAsJSON)
+        }
+      }
+
+      let loginAttempt = defaultState.loginAttempt
+      let preferences = defaultState.preferences
+      let migration = defaultState.migration
+      let tours = defaultState.tours
+      let onboarding = defaultState.onboarding
+
+      await Promise.all([
+        loadLoginAttempt().then((data) => {
+          if (data) {
+            loginAttempt = data
+          }
+        }),
+        loadState<PreferencesState>(LocalStorageKeys.Preferences, (val) => (preferences = val)),
```

Comment:
I had to use the `(val) => (thing = val)` pattern here because I needed to mutate the `thing` variable inside the generic function but I couldn't use `useRef` or `useState` because react would complain about hooks being used in the wrong place